### ISSUE TITLE
Banhammer Isn't working at all on servers in offline mode (Spigot behind an bungee proxy).

### ISSUE THREAD

**Issue by MarioG1** [grandwazir/BanHammer#114]
I have an server network with bungeecord and banhammer on all servers.

The problem is that banned players can join!

The servers (Spigot) behind an bugee proxy get the correct uuid of the players so this can't be the problem.

**Comment by xLRDx** [comment]
Yea i am having this issue as well.

**Comment by xLRDx** [comment]
This is most likely a duplicate of issue #113

the issue is fixable as shown here https://github.com/grandwazir/BanHammer/issues/113#issuecomment-72954744

**Comment by MarioG1** [comment]
I think not that this is related to #113 because its working fine if the server is in online mode ;)

**Comment by xLRDx** [comment]
Sorry my mistake we are sort of both right the issue when fixed brings a new issue to attention and that is the #113 issue. 

It is related to how @grandwazir handles commands. He creates a Async task which the method getPlayer().kick("message") does not support. Get rid of the issue about online/offline mode and its fixed. I have a working version on my server right now running in bungee / spigot with offline mode the other issue (this one) is in PlayerListener the check for 

```
if (!server.getOnlineMode()) return;
```

Im sure there are other ones out there as well but this seems to be working for us

**Comment by grandwazir** [comment]
The reason that this code exists is that it is impossible to rely on any information about players when the server is running in offline mode. When I made that change though I did not consider bungee cord servers. I'll remove it in the next version and instead log a warning to the console if someone is running with offline mode on.

**Comment by xLRDx** [comment]
Yep fully get the idea behind it. I wonder if there is a way to check if the server is running behind a bungee. I know if the server owner chooses to use spigot then they add a setting but that is not reliable either.

**Comment by rustyshakelford2** [comment]
We recently switched to bungeecord and I discovered this bug the hard way. Had to banish a few griefers using /minecraft:ban. Any eta on a fix?

**Comment by rustyshakelford2** [comment]
An update on this would be awesome.
